src/components/Claim/ClaimCreate.tsx

```diff
@@ -122,20 +107,10 @@ const ClaimCreate = ({ personality, claim = {}, sitekey, edit = false }) => {
         }
     }, [content]);
 
-    const onExpiredCaptcha = () => {
-        return new Promise<void>(resolve => {
-            setDisableSubmit(true);
-            resolve();
-        });
-    }
 
-    const onChangeCaptcha = () => {
-        const recaptchaString = recaptchaRef.current.getValue();
-        setRecaptcha(recaptchaString);
+    const onChangeCaptcha = (captchaValid) => {
+        setDisableSubmit(!captchaValid)
```

Comment:
I'm wondering if the is going to be secure. If we do not check during the claim submit we open a vulnerability for the claim to be created by simply sending a Boolean value in the request. I think we can separate the logic but not create a new endpoint for that.

Reply:
@pepermao can help with that if needed, but I think the best solution would be to create a [ValidationPipe](https://docs.nestjs.com/techniques/validation#using-the-built-in-validationpipe) for the DTO that does the recaptcha check.

Reply:
Considering the problems we had with injecting configService in a validationPipe, see if we can use the captcha service approach that I tried on next commit.